src/Idris/Core/TT.hs

```diff
@@ -1674,7 +1674,7 @@ pureTerm (Bind n b sc) = notClassName n && pureBinder b && pureTerm sc where
     pureBinder (Let t v) = pureTerm t && pureTerm v
     pureBinder t = pureTerm (binderTy t)
 
-    notClassName (MN _ c) | c == txt "__class" = False
+    notClassName (MN _ c) | c == txt "__interface" = False
```

Comment:
i just thought to change all the `class` in strings in this PR and all the `class` in docs and code level next PRs respectively.